algebird-test/src/main/scala/com/twitter/algebird/ApproximateProperty.scala

```diff
@@ -0,0 +1,96 @@
+package com.twitter.algebird
+
+import org.scalacheck.{ Gen, Prop }
+import org.scalacheck.util.Pretty
+
+trait ApproximateProperty {
+  type Exact
+  type Approx
+  type Input
+  type Result
+
+  def exactGenerator: Gen[Exact]
+  def inputGenerator(e: Exact): Gen[Input]
+
+  def makeApproximate(e: Exact): Approx
+
+  def exactResult(e: Exact, i: Input): Result
+  def approximateResult(a: Approx, i: Input): Approximate[Result]
+}
+
+object ApproximateProperty {
+
+  /**
+   *  Generates a list of exactly n Ts.
+   *  Useful because `Gen.listOfN(n, gen).sample` gives us Option[List[T]],
+   *  while we often want List[T].
+   */
+  private def genListOf[T](n: Int, gen: Gen[T]): List[T] = {
+    Gen.listOfN(n, gen).sample match {
```

Comment:
I'm concerned that this is not using the Gen.Parameters, so I'm afraid that when scalacheck gives us an example input that fails a test, we can't check it at the REPL. Is there some way to plumb through the Random instance from Gen.Parameters (I think has it), so that we are not using a time-seeded Random?